### ISSUE TITLE
telepresence --new-deployment fails on OpenShift

### ISSUE THREAD

**Issue by cintyaAguirre** [telepresenceio/telepresence#1010]
### What were you trying to do?

(please tell us)

### What did you expect to happen?

(please tell us)

### What happened instead?

(please tell us - the traceback is automatically included, see below.
use https://gist.github.com to pass along full telepresence.log)

### Automatically included information

Command line: `['/usr/bin/telepresence', '--docker-run', '--rm', '-it', 'pstauffer/curl', 'curl', 'http://hello-world:8000/']`
Version: `0.99`
Python version: `3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]`
kubectl version: `Client Version: v1.14.1 // Server Version: v1.11.0+d4cacc0`
oc version: `oc v3.11.0+0cbc58b // kubernetes v1.11.0+d4cacc0 // features: Basic-Auth GSSAPI Kerberos SPNEGO //  // Server https://192.168.42.149:8443 // kubernetes v1.11.0+d4cacc0`
OS: `Linux chinchulin 4.18.0-17-generic #18~18.04.1-Ubuntu SMP Fri Mar 15 15:27:12 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

```
Traceback (most recent call last):
  File "/usr/bin/telepresence/telepresence/cli.py", line 136, in crash_reporting
    yield
  File "/usr/bin/telepresence/telepresence/main.py", line 60, in main
    remote_info = start_proxy(runner)
  File "/usr/bin/telepresence/telepresence/proxy/__init__.py", line 95, in start_proxy
    run_id=run_id,
  File "/usr/bin/telepresence/telepresence/proxy/remote.py", line 153, in get_remote_info
    runner, deployment_name, deployment_type, run_id=run_id
  File "/usr/bin/telepresence/telepresence/proxy/remote.py", line 100, in get_deployment_json
    )["items"][0]
IndexError: list index out of range

```

Logs:

```
ound svc,deploy --selector=telepresence=34313c292d7048b2a6e71951d1a8b7de
   8.3  11 | No resources found
   8.3 TEL | [11] ran in 0.13 secs.
   8.3 TEL | [12] Running: oc --context minishift --namespace myproject run --restart=Always --limits=cpu=100m,memory=256Mi --requests=cpu=25m,memory=64Mi telepresence-1556205670-4381108-32764 --image=datawire/telepresence-k8s:0.99 --labels=telepresence=34313c292d7048b2a6e71951d1a8b7de
   8.5  12 | deploymentconfig.apps.openshift.io/telepresence-1556205670-4381108-32764 created
   8.5 TEL | [12] ran in 0.13 secs.
   8.5 TEL | END SPAN deployment.py:70(create_new_deployment)    0.3s
   8.5 TEL | BEGIN SPAN remote.py:151(get_remote_info)
   8.5 TEL | BEGIN SPAN remote.py:78(get_deployment_json)
   8.5 TEL | [13] Capturing: oc --context minishift --namespace myproject get deployment -o json --selector=telepresence=34313c292d7048b2a6e71951d1a8b7de
   8.6 TEL | [13] captured in 0.12 secs.
   8.6 TEL | END SPAN remote.py:78(get_deployment_json)    0.1s

```

**Comment by ark3** [comment]
Sorry about the crash! It appears that the new deployment operation is broken on OpenShift. Telepresence uses `kubectl run` to launch its proxy, which creates a DeploymentConfig, but then proceeds to treat that as a Deployment, which means connect and cleanup both fail.

Telepresence should construct a manifest and use `kubectl apply` so it knows exactly what it is getting. Doing that would also fix #494.

Until we fix this, please give the [swap deployment workflow](https://www.telepresence.io/tutorials/openshift) a try.

**Comment by ark3** [comment]
See also #949.

**Comment by ark3** [comment]
I believe this was fixed by #1404.
